### PR TITLE
Replay #2221: Implement GitHub Actions Secrets API for both Organization and Repository

### PR DESCRIPTION
**Pull request by kfcampbell**
This PR resurrects @mptolly-takeda's branch in #2221 before the erroneous merge conflict.

### CODE REVIEW

**Comment by kfcampbell** [comment]
Note: to resolve the merge conflict here, I took everything in the .csproj file from the main branch.

File: Octokit/Models/Response/OrganizationSecret.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            CreatedAt = createdAt;
+            UpdatedAt = updatedAt;
+            Visibility = visibility;
+            SelectedRepositoriesUrl = selectedRepositoriesUrl;
+        }
+
+        /// <summary>
+        /// The visibility level of the secret within the organization
+        /// </summary>
+        public string Visibility { get; protected set; }
```
Comment (JonruAlveus) [inline Octokit/Models/Response/OrganizationSecret.cs:31]: We’ve recently changed to private setters for response models. 👍

Comment (kfcampbell) [reply Octokit/Models/Response/OrganizationSecret.cs]: I've changed to private setters in all introduced models! Thanks.

Comment (kfcampbell) [reply Octokit/Models/Response/OrganizationSecret.cs]: Okay, not quite _all_ introduced response models. `OrganizationSecret` inherits from `RepositorySecret` so `RepositorySecret` needs protected setters.

Comment (JonruAlveus) [reply Octokit/Models/Response/OrganizationSecret.cs]: 👍

**Review by JonruAlveus — APPROVED** [review]

**Review by nickfloyd — APPROVED** [review]
💥

**Pull request merged**